index.js

```diff
@@ -17,7 +19,13 @@ module.exports = function (str, filepath) {
 
 	return pify(mkdirp, Promise)(path.dirname(fullpath))
 		.then(function () {
-			return pify(fs.writeFile, Promise)(fullpath, str);
+			return isStream(str) ?
+				new Promise((resolve, reject) => {
+					str.pipe(fs.createWriteStream(fullpath))
```

Comment:
> Me too. whatwg/streams can't come soon enough.

👍 👍 👍 


---

> If they are trapped for any reason, would my writable stream hang on to its file descriptor forever?

From the readable stream docs:

> If autoClose is set to true (default behavior), on error or end the file descriptor will be closed automatically.

So it seems it's handled for us even if we listen to `error`. So seems fine to just reject, right? I'm good with both solutions though, so up to you.